<?xml version="1.0"?>
<launch>
    <node name="processImage" pkg="charlie" type="processImage" output="screen" clear_params="true">
        <param name="process_scale" value="0.25" />
        <param name="show_images" value="False" />
        <param name="check_intersections" value="False" />
        <param name="use_pid" value="False" />

        <param name="b_lower" value="0" />
        <param name="g_lower" value="0" />
        <param name="r_lower" value="0" />

        <param name="b_upper" value="255" />
        <param name="g_upper" value="255" />
        <param name="r_upper" value="255" />

        <param name="h_lower" value="80" />
        <param name="s_lower" value="80" />
        <param name="v_lower" value="0" />

        <param name="h_upper" value="140" />
        <param name="s_upper" value="255" />
        <param name="v_upper" value="255" />
    </node>
    
    <!--<node name="controller" pkg="charlie" type="controller_with_gui.py" output="screen" clear_params="true">
    </node>-->
    
    <node name="controller" pkg="charlie" type="create_eyes_controller.py" output="screen" clear_params="true">
    </node> 

    <node name="sonarPublisher" pkg="charlie" type="sonarNode.py" output="screen" clear_params="true">
        <param name="sampleParam" value="none" />
    </node> 
 

  <node name="beeper" pkg="charlie" type="beepNode.py" output="screen" clear_params="true">
      <param name="beepType" value="continuous" />
  </node> 

  <arg name="name" default="raspicam_node" />

  <!-- Raspicam Node -->

  <node type="raspicam_node" pkg="raspicam_node" name="$(arg name)" output="screen">

    <param name="camera_info_url" value="package://$(arg name)/camera_info/camera_module_v2_640x480.yaml"/>
    <param name="width" value="640"/>
    <param name="height" value="480"/>
    <param name="framerate" value="30"/>
    <param name="camera_frame_id" value="raspicam"/>

  </node>

  <node pkg="rosservice" type="rosservice" name="start_capture" args="call --wait $(arg name)/start_capture"/>

  <arg name="config" default="$(find ca_driver)/config/default.yaml" />
  <arg name="desc" default="true" />

  <node name="ca_driver" pkg="ca_driver" type="ca_driver" output="screen">
    <rosparam command="load" file="$(arg config)" />
    <param name="robot_model" value="CREATE_2" />
  </node>

  <!-- Robot description -->
  <include if="$(arg desc)" file="$(find ca_description)/launch/create_2.launch" />

</launch>

